telemetry/telemetry.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package telemetry
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/types"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const (
+	telemetryPkgPath = "github.com/mattermost/mattermost-server/v5/services/telemetry"
+	modelPkgPath     = "github.com/mattermost/mattermost-server/v5/model"
+)
+
+// Analyzer describes telemetry status analysis function for a config setting
+var Analyzer = &analysis.Analyzer{
+	Name: "telemetry",
+	Doc:  "reports if a config field is not used in services/telemetry package",
```

Comment:
Although this description is pedantically correct, I think it would be more useful if it actually conveyed the _intent_ of the analyser to someone reading it. Maybe this instead:

```suggestion
	Doc:  "reports config fields where telemetry is not sent (unless it is specifically indicated  that no telemetry will be sent for that field)",
```

Reply:
Sounds good to me.